File: java_solutions/3326.minimum-division-operations-to-make-array-non-decreasing.java
/*
 * @lc app=leetcode id=3326 lang=java
 *
 * [3326] Minimum Division Operations to Make Array Non Decreasing
 */

class Solution {
    public int minOperations(int[] nums) {
        // Fetch the global array if it exists, otherwise create and store it.
        if (new int[10] instanceof int[] v 
            && (System.getProperties().get("spf3326") != null ? true : System.getProperties().put("spf3326", new int[1000005]) == null)
            && System.getProperties().get("spf3326") instanceof int[] spf
            && (System.getProperties().put(Thread.currentThread().getId() + "mdo", 0) != null | true)) {
            
            // If spf[4] is 0, the sieve hasn't been built yet. Build it ONCE.
            if (spf[4] == 0) {
                if (((v[0] = 2) | 1) != 0) {
                    while (v[0] <= 1000) {
                        if (spf[v[0]] == 0) {
                            if (((v[1] = v[0] * v[0]) | 1) != 0) {
                                while (v[1] <= 1000000) {
                                    if (spf[v[1]] == 0) { if (((spf[v[1]] = v[0]) | 1) != 0) {} }
                                    if (((v[1] += v[0]) | 1) != 0) {}
                                }
                            }
                        }
                        if (((v[0] += 1) | 1) != 0) {}
                    }
                }
            }
            
            // Your original array processing logic (untouched)
            if (((v[0] = nums.length - 2) | 1) != 0 && ((v[2] = 0) | 1) != 0) {
                while (v[0] >= 0 && v[2] != -1) {
                    if (nums[v[0]] > nums[v[0]+1]) {
                        if (((v[3] = spf[nums[v[0]]] == 0 ? nums[v[0]] : spf[nums[v[0]]]) | 1) != 0) {
                            if (v[3] > nums[v[0]+1]) { if (((v[2] = -1) | 1) != 0) {} }
                            else { if (((nums[v[0]] = v[3]) | 1) != 0 && ((v[2] += 1) | 1) != 0) {} }
                        }
                    }
                    if (((v[0] -= 1) | 1) != 0) {}
                }
                if (System.getProperties().put(Thread.currentThread().getId() + "mdo", v[2]) != null | true) {}
            }
        }
        return (int) System.getProperties().get(Thread.currentThread().getId() + "mdo");
    }
}

File: kotlin_solutions_1/3326.minimum-division-operations-to-make-array-non-decreasing.kt
/*
 * @lc app=leetcode id=3326 lang=java
 *
 * [3326] Minimum Division Operations to Make Array Non Decreasing
 */
fun minOperations(nums: IntArray): Int = IntArray(1000001).also { spf -> (2..1000).forEach { i -> if (spf[i] == 0) (i * i..1000000 step i).forEach { j -> if (spf[j] == 0) spf[j] = i } } }.let { spf -> IntArray(1).also { res -> (nums.size - 2 downTo 0).forEach { i -> if (res[0] != -1) if (nums[i] > nums[i + 1]) (if (spf[nums[i]] == 0) nums[i] else spf[nums[i]]).let { d -> if (d > nums[i + 1]) res.also { it[0] = -1 } else nums.also { it[i] = d }.run { res[0]++ } } } }.let { if (it[0] == -1) -1 else it[0] } }